tests/stages/arxiv/test_arxiv_source.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+# SPDX-FileCopyrightText: Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import string
+import types
+from unittest import mock
+
+import pytest
+
+import cudf
+
+from _utils.dataset_manager import DatasetManager
+from morpheus.config import Config
+from morpheus.stages.input.arxiv_source import ArxivSource
+
+
+@mock.patch("langchain.text_splitter.RecursiveCharacterTextSplitter")
+def test_constructor(mock_splitter: mock.MagicMock, config: Config):
+    cache_dir = "/does/not/exist"
+    stage = ArxivSource(config, query="unittest", cache_dir=cache_dir, chunk_size=77, chunk_overlap=33, max_pages=100)
+    assert stage._query == "unittest"
+    assert stage._cache_dir == cache_dir
+    assert stage._max_pages == 100
+
+    mock_splitter.assert_called_once_with(chunk_size=77, chunk_overlap=33, length_function=len)
+
+    assert not os.path.exists(cache_dir)
+
+
+@pytest.mark.parametrize("chunk_size,chunk_overlap", [(99, 100), (100, 100)])
+def test_constructor_chunk_size_error(config: Config, chunk_size: int, chunk_overlap: int):
+    with pytest.raises(ValueError):
+        ArxivSource(config, query="unittest", chunk_size=chunk_size, chunk_overlap=chunk_overlap)
+
+
+def _make_mock_result(file_name: str):
+    result = mock.MagicMock()
+    result._get_default_filename.return_value = file_name
+    return result
+
+
+@pytest.mark.parametrize("use_subdir", [False, True])
+def test_generate_frames_cache_miss(mock_arxiv_search: mock.MagicMock, config: Config, tmp_path: str, use_subdir: bool):
+    if use_subdir:
+        # Tests that the cache directory is created if it doesn't exist
+        cache_dir = os.path.join(tmp_path, "cache")
+        assert not os.path.exists(cache_dir)
+    else:
+        cache_dir = tmp_path
+
+    stage = ArxivSource(config, query="unittest", cache_dir=cache_dir)
+
+    expected_file_paths = [os.path.join(cache_dir, "apples.pdf"), os.path.join(cache_dir, "plums.pdf")]
+    assert list(stage._generate_frames()) == expected_file_paths
+
+    assert os.path.exists(cache_dir)
+    assert stage._total_pdfs == 2
```

Comment:
Same comment here; if 'total_pdfs' is a value that we want to expose, we should do it and make it clear that its part of the public interface.